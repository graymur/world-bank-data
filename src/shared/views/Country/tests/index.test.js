/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';
import {Country} from '../';
import _ from 'lodash';

const props = {
	match: {
		params: {iso2Code: 'AA'}
	},
	loading: false,
	country: {
		iso2Code: 'AA',
		name: 'Country name',
		region: {value: 'Region name'},
		incomeLevel: {value: 'Income level'},
		capitalCity: 'City name'
	},
	indicators: [{
		id: '1',
		name: 'First indicator'
	}, {
		id: '2',
		name: 'Second indicator'
	}]
};

test('Renders country and indicators list', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Country {...props}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Calls loadCountry prop if country is not set', () => {
	const thisProps = _.omit(props, 'country');

	thisProps.loadCountry = jest.fn();

	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Country {...thisProps}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	expect(thisProps.loadCountry.mock.calls.length).toBe(1);
	expect(thisProps.loadCountry.mock.calls[0][0]).toBe('AA');
});

test('Calls loadIndicators prop if indicators are not set', () => {
	const thisProps = _.omit(props, 'indicators');

	thisProps.indicators = [];
	thisProps.loadIndicators = jest.fn();

	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Country {...thisProps}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	expect(thisProps.loadIndicators.mock.calls.length).toBe(1);
});
