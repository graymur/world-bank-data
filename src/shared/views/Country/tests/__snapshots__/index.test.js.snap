// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calls loadIndicators prop if indicators are not set 1`] = `
<div
  className="country"
>
  <div
    className="country"
  >
    <h1>
      Country name
    </h1>
    <p>
      Region: 
      Region name
    </p>
    <p>
      Income level: 
      Income level
    </p>
    <p>
      Capital city: 
      City name
    </p>
    <h1>
      Indicators
    </h1>
    <nav
      className="indicators__list"
    />
  </div>
</div>
`;

exports[`Renders country and indicators list 1`] = `
<div
  className="country"
>
  <div
    className="country"
  >
    <h1>
      Country name
    </h1>
    <p>
      Region: 
      Region name
    </p>
    <p>
      Income level: 
      Income level
    </p>
    <p>
      Capital city: 
      City name
    </p>
    <h1>
      Indicators
    </h1>
    <nav
      className="indicators__list"
    >
      <a
        className="indicators__list__item"
        href="/countries/AA/indicator/1"
        onClick={[Function]}
      >
        First indicator
      </a>
      <a
        className="indicators__list__item"
        href="/countries/AA/indicator/2"
        onClick={[Function]}
      >
        Second indicator
      </a>
    </nav>
  </div>
</div>
`;
